tests/test_url.py

```diff
@@ -647,6 +647,11 @@ def test_with_query():
     assert str(url.with_query({'a': '1'})) == 'http://example.com/?a=1'
 
 
+def test_with_query_kwargs():
```

Comment:
test should demonstrate query replacement, not adding a query to empty querystring.